src/api/MainSolver.h

```diff
@@ -185,6 +186,20 @@ class MainSolver {
         PTRef result {PTRef_Undef};
     };
 
+    class Preprocessor {
+        Substitutions substitutions;
+        opensmt::ScopedVector<PTRef> preprocessedFormulas;
+    public:
+        void push();
+        void pop();
+        void initialize();
+        void addProcessedFormula(PTRef);
+        [[nodiscard]] opensmt::span<const PTRef> getPreprocessedFormulas() const;
+        [[nodiscard]] Logic::SubstMap getCurrentSubstitutions() const { return substitutions.current(); }
+        void setSubstitutions(std::size_t level, Logic::SubstMap && subs) { substitutions.set(level, std::move(subs)); }
```

Comment:
I disagree. `Substitutions` is a class with its own public API and I don't think it should be changed because of how `Preprocessor` wants to use this class.

Reply:
OK, my impression was that it turned into just a helper class inside of `Preprocessor`.